src/solvers/qbf/qdimacs_core.cpp

```diff
@@ -58,8 +58,7 @@ void qdimacs_coret::simplify_extractbits(exprt &expr) const
             if(oit->op0().get(ID_identifier)==ident)
             {
               const exprt &val_expr=oit->op1();
-              mp_integer value;
-              to_integer(val_expr, value);
+              mp_integer value = numeric_cast_v<mp_integer>(val_expr);
```

Comment:
Since `value` is converted to ulong just afterwards it may make sense to directly cast to `unsigned long` here.

Reply:
also could be const